modules/platform/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/adapters/AsyncOperationAdapter.java

```diff
@@ -333,11 +334,38 @@ protected void setError(String executionId, Throwable t) {
 
     public int getErrorStatus(String executionId) {
         Long status = (Long) getTransientStore().getParameter(executionId, TRANSIENT_STORE_ERROR_STATUS);
-        return status == null ? 0 : status.intValue();
+        if (status != null) {
+            return status.intValue();
+        }
+
+        // check the errors, if the execution is a bulk action
+        if (isAsync(executionId)) {
+            Serializable taskId = getAsyncTaskId(executionId);
```

Comment:
Checking `isAsync` before calling `getAsyncTaskId` is redundant. The new pattern in this code after the latest refactoring/cleanup is just:

            Serializable taskId = getAsyncTaskId(executionId);
            if (taskId != null) {
                ...
            }